robotools/tests.py

```diff
@@ -247,66 +247,79 @@ def test_context(self):
         
         with self.assertRaises(ValueError):
-            evotools._prepate_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, liquid_class=None)
+            evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, liquid_class=None)
         with self.assertRaises(ValueError):
-            evotools._prepate_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, liquid_class='liquid;class')
-        evotools._prepate_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, liquid_class='valid liquid class')
+            evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, liquid_class='liquid;class')
+        evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, liquid_class='valid liquid class')
         with self.assertRaises(ValueError):
-            evotools._prepate_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=None)
+            evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=None)
         with self.assertRaises(ValueError):
-            evotools._prepate_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=12)
-        evotools._prepate_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=4)
-        evotools._prepate_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=evotools.Tip.T5)
+            evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=12)
         with self.assertRaises(ValueError):
-            evotools._prepate_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, rack_id=None)
+            evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=[1, 2.6])
+        evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=4)
+
+        _, _, _, _, tip, _, _, _, _ = evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=[evotools.Tip.T1, 4])
+        _, _, _, _, tip2, _, _, _, _ = evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=[1, 4])
+        _, _, _, _, tip3, _, _, _, _ = evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=[1, evotools.Tip.T4])
+        self.assertEqual(tip, tip2)
+        self.assertEqual(tip2, tip3)
         with self.assertRaises(ValueError):
-            evotools._prepate_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, rack_id='invalid;rack')
+            evotools._prepare_aspirate_dispense_parameters(rack_label='WaterTrough', position=1, volume=15, tip=(evotools.Tip.T1, evotools.Tip.Any))
```

Comment:
Also here please don't follow my bad example, but match the message with `pytest.raises(, match=...)`

Reply:
Done